track2 .NET SDK generation for dataprotection

hiaga opened the pull request:
# Contributing to the Azure SDK

Please see our [CONTRIBUTING.md](https://github.com/Azure/azure-sdk-for-net/blob/main/CONTRIBUTING.md) if you are not familiar with contributing to this repository or have questions.

For specific information about pull request etiquette and best practices, see [this section](https://github.com/Azure/azure-sdk-for-net/blob/main/CONTRIBUTING.md#pull-request-etiquette-and-best-practices).

Yao725 commented on sdk/dataprotection/Azure.ResourceManager.DataProtectionBackup/src/Customized/Models/DataProtectionBackupVaultProperties.cs:
```diff
@@ -0,0 +1,14 @@
+    public partial class DataProtectionBackupVaultProperties
+    {
+        /// <summary> Is vault protected by resource guard. </summary>
+        public bool? IsVaultProtectedByResourceGuard { get; set; }
```
This property is marked as `readOnly: true` in the API spec after our last stable release. So the latest generated code will remove the setter. I manually add setter back but we will not send it to service as it is removed from the serialization.

azure-sdk commented:
**API change check**

APIView has identified API level changes in this PR and created following API reviews.

[Azure.ResourceManager.DataProtectionBackup](https://apiview.dev/Assemblies/Review/ad8e99c86f0746d6a9c20e0d76620d41)

hiaga committed:
change the type of subscription id
fix breaking changes and update api
regen code based on latest swagger 2023-01-01

hiaga commented:
@m-nash could you please review this PR

hiaga committed:
added back DataStoreParametersList.Get()

hiaga committed:
regen code

hiaga committed:
added DataStoreParametersList EB.Never

hiaga committed:
Added changeLog

Yao725 commented on sdk/dataprotection/Azure.ResourceManager.DataProtectionBackup/src/Customized/Models/BackupInstancePolicyInfo.cs:
```diff
@@ -0,0 +1,37 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+// <auto-generated/>
```
remove this line

hiaga replied:
done

Yao725 commented on sdk/dataprotection/Azure.ResourceManager.DataProtectionBackup/CHANGELOG.md:
```diff
@@ -10,6 +10,17 @@
 
 ### Other Changes
 
+## 1.1.0 (2023-01-30)
```
Please give two weeks buffer when trying to specify the release date.
Please remove the content of 1.1.0-beta.1 above.

hiaga replied:
removed and updated release date as discussed

Yao725 commented on sdk/dataprotection/Azure.ResourceManager.DataProtectionBackup/CHANGELOG.md:
```diff
@@ -10,6 +10,17 @@
 ### Other Changes
 
+## 1.1.0 (2023-01-30)
+This release adds support for mutliple features for Microsoft Azure Data Protection Backup management client library.
```
Please remove this line.

hiaga commented on sdk/dataprotection/Azure.ResourceManager.DataProtectionBackup/src/Generated/Models/RestoreFilesTargetDetails.cs:
```diff
@@ -46,5 +46,11 @@ public RestoreFilesTargetDetails(string filePrefix, RestoreTargetLocationType re
+        /// This is optional if the target subscription can be identified with the URL field. If not
+        /// then this is needed if CrossSubscriptionRestore field of BackupVault is in any of the disabled states
+        /// </summary>
+        public string TargetResourceArmId { get; set; }
```
yes this will be the complete ARM ID of the target resource, similar to what you mentioned. will check on the exact format

hiaga replied:
checked, It is complete ARM ID of the target resource and can be of different types like bolbs, databases etc.

hiaga committed:
resolved review comments

cvarsh commented on sdk/dataprotection/Azure.ResourceManager.DataProtectionBackup/CHANGELOG.md:
```diff
@@ -1,24 +1,20 @@
 # Release History
 
-## 1.1.0-beta.1 (Unreleased)
+## 1.1.0 (2023-01-04)
```
Is this correct? or should be 2023-02-04?

Yao725 committed:
Merge remote-tracking branch 'upstream/main' into hiaga/dppSDK

Yao725 committed:
update release date and changelog

Yao725 reviewed (approved):
LGTM, will polish the API in another separate PR.
